tools/tinytest-codegen.py

```diff
@@ -112,11 +130,20 @@ def script_to_map(test_file):
 )
 argparser.add_argument("--stdin", action="store_true", help="read list of tests from stdin")
 argparser.add_argument("--exclude", action="append", help="exclude test by name")
+argparser.add_argument(
+    "--profile",
+    type=argparse.FileType("rt", encoding="utf-8"),
+    help="optional profile file providing test directories and exclusion list",
+)
 args = argparser.parse_args()
 
 if not args.stdin:
+    if args.profile:
+        directories, excluded = load_profile(args.profile)
```

Comment:
We're using different names for the same thing, e.g. (directories, test_dirs), (exclude_tests, excluded, exclusion_list).

Suggest:

```
profile_test_dirs, profile_exclude_tests = load_profile(args.profile)
test_dirs = test_dirs.union(profile_test_dirs)
exclude_tests = exclude_tests.union(profile_exclude_tests)
```